src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -680,24 +798,120 @@ def get_observation(self, event: Event) -> pd.DataFrame:
             | (df_eligible_dependents["last_year_income"] < 4_300)
         ]
 
-        s_eligible_dependent_id_list = df_eligible_dependents.groupby("guardian_id").apply(
-            lambda df_g: list(df_g.index)
-        )
+        df = df_eligible_dependents
+        df["dependent_id"] = df.index
+        df["tax_year"] = event.time.year - 1
 
-        s_eligible_dependent_address_id_list = df_eligible_dependents.groupby(
-            "guardian_id"
-        ).apply(lambda df_g: list(df_g["address_id"]))
+        # merge in simulant columns based on simulant id
+        for col in [
+            "first_name_id",
+            "middle_name_id",
+            "last_name_id",
+            "age",
+            "date_of_birth",
+            "sex",
+            "ssn",
+            "address_id",
+            "housing_type",
+        ]:
+            df[col] = df["dependent_id"].map(pop_full[col])
 
-        for col, s in [
-            ["eligible_dependent_id_list", s_eligible_dependent_id_list],
-            ["eligible_dependent_address_id_list", s_eligible_dependent_address_id_list],
+        # delete extraneous columns
```

Comment:
Much cleaner IMO to specify the columns that you _do_ want than to remove the ones you don't. I believe you could use `OUTPUT_COLUMNS` for this, as well.

Reply:
> Much cleaner IMO to specify the columns that you do want than to remove the ones you don't. I believe you could use OUTPUT_COLUMNS for this, as well.

Agree, I'll change to that.